WordPress/src/main/java/org/wordpress/android/util/LocaleManager.java

```diff
@@ -0,0 +1,151 @@
+package org.wordpress.android.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.os.Build;
+import android.preference.PreferenceManager;
+
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ * Helper class for working with localized strings. Ensures updates to the users
+ * selected language is properly saved and resources appropriately updated for the
+ * android version.
+ */
+public abstract class LocaleManager {
+    private static final String LANGUAGE_KEY = "language-pref";
+
+    /**
+     * Activate the locale associated with the provided context.
+     * @param context The current context.
+     */
+    public static Context setLocale(Context context) {
+        return updateResources(context, getLanguage(context));
+    }
+
+    /**
+     * Change the active locale to the language provided. Save the updated language
+     * settings to sharedPreferences.
+     * @param context The current context
+     * @param language The 2-letter language code (example "en") to switch to
+     */
+    public static Context setNewLocale(Context context, String language) {
+        Locale newLocale = WPPrefUtils.languageLocale(language);
+
+        if (Locale.getDefault().toString().equals(newLocale.toString())) {
+            removePersistedLanguage(context);
+        } else {
+            saveLanguageToPref(context, language);
+        }
+        return updateResources(context, language);
+    }
+
+    /**
+     * Compare the language for the current context with another language.
+     * @param language The language to compare
+     * @return True if the languages are the same, else false
+     */
+    public static boolean isDifferentLanguage(String language) {
```

Comment:
It might just be me, but I think it'd be better if we had a `isSameLanguage` method instead of `isDifferentLanguage` to avoid the double negative like the one in `AppSettingsFragment`:

```
if (!LocaleManager.isDifferentLanguage(languageCode)) {
    return;
}
```

What do you think?

Reply:
Love it...I'll get it updated 💯